### PR TITLE
[Inductor] Fix fused argmax/argmin index on channels_last

### LINKED ISSUE #193751: [inductor] Fused reduction + argmax/argmin mis-indexes `channels_last` inputs (logical index not preserved), CPU and CUDA
### 🐛 Describe the bug

`torch.compile` returns a wrong `argmax`/`argmin` index when a trailing-dim reduction (`mean`, `sum`, `amax`, `var`, `logsumexp`, `prod`) is fused with the index consumer and the input uses the `channels_last` memory format (4D, non-contiguous). Eager is correct; the compiled index points at a value that is not the extreme of the reduction output. The generated kernel computes the reduction values correctly but reports the physical flat index of a non-row-major intermediate buffer instead of the logical row-major index.

### To reproduce

```python
import torch
torch.manual_seed(1234)
x = torch.randn(4, 4, 8, 8).to(memory_format=torch.channels_last)  # strides (256, 1, 32, 4)

def fn(t):
    return torch.argmax(torch.mean(t, dim=-1))

print("eager   =", fn(x).item())                # 14 (correct, value 0.751697 == max)
print("compiled=", torch.compile(fn)(x).item()) # 25 (wrong, value -0.410133)
```

All six ops fail the same way (2.13 CPU, seed 1234; nightly 2.15.0.dev20260816+cu130 reproduces with identical indices on CPU and CUDA):

| op | eager idx -> value | compiled idx -> value | extreme |
|---|---|---|---|
| mean + argmax | 14 -> 0.751697 | 25 -> -0.410133 | 0.751697 |
| sum + argmax | 14 -> 6.013576 | 25 -> -3.281064 | 6.013576 |
| amax + argmax | 28 -> 3.029174 | 19 -> 0.945478 | 3.029174 |
| var + argmax | 43 -> 2.989961 | 45 -> 0.611232 | 2.989961 |
| logsumexp + argmax | 54 -> 3.537204 | 58 -> 2.333352 | 3.537204 |
| prod + argmin | 34 -> -2.680438 | 40 -> 5.749767 | -2.680438 |

`channels_last_3d` (5D, NHWDC) reproduces identically, e.g. mean+argmax: eager 232 -> 1.272395, compiled 163 -> 0.385751 (2.13 CPU, seed 1234; reproduces on nightly and CUDA too).

Controls pass: contiguous input, `[..., ::2]`, and `expand`/stride-0 views; only flat `argmax`/`argmin` consumers are affected (`max`/`min` dim and `topk` consumers pass).

### Root cause (kernel evidence, 2.13 CPU)

The fused kernel `cpp_fused_argmax_mean_0` reads the input with correct channels_last strides and computes the mean correctly, but the argmax stage reports the flat loop position as the index while the intermediate mean buffer has strides `(32, 1, 4)` (non-row-major): physical position 25 corresponds to logical row-major position 14. The candidate value is correct; only the reported index is wrong.

### Why this is still open

The root cause is that the fused arg-reduction reports the physical flat index of a non-row-major intermediate buffer instead of the logical row-major index. We observed a producer-type split: variants whose arg-reduction input is an unrealized pointwise result (mean/var/logsumexp) can be made correct by enabling the logical-index path, while variants whose input is a materialized, contiguously-allocated reduction buffer (sum/amax/prod) remain wrong under the same change — the existing layout check does not fire for a contiguous allocation. Latest nightly (2.15.0.dev20260816+cu130) still reproduces all six variants on CPU and CUDA, so nothing in the current tree fixes this.

### Expected behavior

`torch.compile` should return the same logical row-major index as eager for channels_last inputs across all fused reduction ops.

### Versions

```text
PyTorch version: 2.13.0+cu130 and nightly 2.15.0.dev20260816+cu130 (both reproduce, CPU and CUDA)
OS: Linux (AutoDL container)
```

cc @jamesr66a @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @aakhundov @coconutruben @jataylo

### PR DESCRIPTION
**Pull request by ShravanthReddy**
Fused argmax/argmin was returning the physical loop index of a non-row-major buffer, so `channels_last` inputs got the wrong index.

Always report the logical row-major index when flattening more than one dim.

Fixes #193751

Test plan (CPU nightly, seed 1234):

official script:
| | eager | compiled |
| unpatched | 14 | 25 |
| patched | 14 | 14 |

compiled matched eager after the patch for:
- mean/sum/amax/var/logsumexp + argmax on channels_last
- prod + argmin on channels_last
- channels_last_3d mean+argmax
- contiguous NCHW control
- aot_eager
- bare argmax/argmin
- contiguous reduction+argreduce
- channels_last with dim != -1

`python test/inductor/test_channels_last_fused_argreduce.py`

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

### CODE REVIEW

**Comment by jansel** [comment]
Fixing the same issue as #195063 -- I think the fix over there is more narrow than this one.
